### PR TITLE
Add unit tests

### LINKED ISSUE #49: Setup tests
We need to setup unit tests for the react components under /webviews/

### PR DESCRIPTION
**Pull request by neehar18**
Fixes #49

### CODE REVIEW

**Comment by fbricon** [comment]
can you update the CI workflow so react tests are run?

**Comment by neehar18** [comment]
@fbricon can you explain what you meant by updating the CI workflow?

**Comment by fbricon** [comment]
update https://github.com/redhat-developer/vscode-granite/blob/main/.github/workflows/build.yaml so that tests are run by the CI agent

**Comment by neehar18** [comment]
@fbricon this [line](https://github.com/redhat-developer/vscode-granite/blob/aff28698ff6fe79ec3823f99f3a91c2304b28e93/.github/workflows/build.yaml#L60) in the existing build.yaml should trigger the unit tests that I added

**Comment by fbricon** [comment]
@neehar18 no. You can try running "npm test" at the root of the project and see if your react tests are run

**Comment by fbricon** [comment]
also please fix the conflicts

**Comment by neehar18** [comment]
@fbricon got it. I'll make the necessary changes and update the PR. 

Also, when do you want the tests to be run? Before building extension and generating vsix pkg or after?

**Comment by fbricon** [comment]
after packaging, so we can still test the vsix even if there's a failure

File: test-results.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (fbricon) [inline test-results.xml:1]: Test results don't belong in this repo.

please add it to .gitignore so it's excluded

File: webviews/src/App.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import "./App.css";
 import { useCallback, useEffect, useState } from "react";
 import ModelList, { ModelOption } from "./ModelList";
 import { ProgressData } from "../../src/commons/progressData";
+import React from 'react';
```
Comment (fbricon) [inline webviews/src/App.tsx:6]: seems unnecessary

File: webviews/src/StatusCheck.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import React from "react";
```
Comment (fbricon) [inline webviews/src/StatusCheck.tsx:1]: seems unnecessary

**Comment by neehar18** [comment]
The conflicts in package-lock.json got out of hand. I opened a new PR to address this: https://github.com/redhat-developer/vscode-granite/pull/111

**Pull request closed without merging**